cpp/include/raft/matrix/init.cuh

```diff
@@ -18,28 +18,49 @@
 
 #include <raft/core/device_mdspan.hpp>
 #include <raft/core/host_mdspan.hpp>
-#include <raft/matrix/detail/matrix.cuh>
+#include <raft/matrix/detail/math.cuh>
 #include <raft/matrix/matrix.cuh>
 
 namespace raft::matrix {
 /**
  * @brief set values to scalar in matrix
  * @tparam math_t data-type upon which the math operation will be performed
- * @tparam idx_t integer type used for indexing
+ * @tparam extents dimension and indexing type used for the input
  * @tparam layout layout of the matrix data (must be row or col major)
  * @param[in] handle: raft handle
  * @param[in] in input matrix
  * @param[out] out output matrix. The result is stored in the out matrix
  * @param[in] scalar scalar value to fill matrix elements
  */
-template <typename math_t, typename idx_t, typename layout>
+template <typename math_t, typename extents, typename layout>
 void fill(const raft::handle_t& handle,
-          raft::device_matrix_view<const math_t, idx_t, layout> in,
-          raft::device_matrix_view<math_t, idx_t, layout> out,
+          raft::device_mdspan<const math_t, extents, layout> in,
+          raft::device_mdspan<math_t, extents, layout> out,
           raft::host_scalar_view<math_t> scalar)
 {
+  RAFT_EXPECTS(raft::is_row_or_column_major(out), "Data layout not supported");
   RAFT_EXPECTS(in.size() == out.size(), "Input and output matrices must be the same size.");
+  RAFT_EXPECTS(scalar.data_handle() != nullptr, "Empty scalar");
   detail::setValue(
     out.data_handle(), in.data_handle(), *(scalar.data_handle()), in.size(), handle.get_stream());
 }
+
+/**
+ * @brief set values to scalar in matrix
+ * @tparam math_t data-type upon which the math operation will be performed
+ * @tparam extents dimension and indexing type used for the input
+ * @tparam layout_t layout of the matrix data (must be row or col major)
+ * @param[in] handle: raft handle
+ * @param[inout] inout input matrix
+ * @param[in] scalar scalar value to fill matrix elements
+ */
+template <typename math_t, typename extents, typename layout>
+void fill(const raft::handle_t& handle,
+          raft::device_mdspan<math_t, extents, layout> inout,
+          math_t scalar)
```

Comment:
This was intended to be a much easier version to use because `scalar` is not used as a "out parameter". I don't see the need for it to be host scalar view.
Otherwise we'd have to create a host scalar view that holds a pointer to zero every time we want to reset to zero a matrix?
And the `in` parameter seems to be useless in the other overload as well when looking at the implementation details.

Reply:
I'm actually thinking we can probably hold off on adding the host scalar view version altogether for now until we have a need to support the device scalar view. These are more generally useful when we have a pointer to device memory that we want to use as input but we don't want to synchronize the stream each time.